WordPress/Classes/ViewRelated/Gutenberg/Collapsable Header/Collapsable Header Collection View Cell/CollapsableHeaderCollectionViewCell.swift

```diff
@@ -21,6 +21,9 @@ class CollapsableHeaderCollectionViewCell: UICollectionViewCell {
         }
     }
 
+    /// The throttle the requests to the imageURL polling if needed.
+    private let throttle = Scheduler(seconds: 1)
```

Comment:
I set the throttle to 1 second here. This has been resulting in about 5 or 6 attempts before the image successfully loads. I'm not sure what the right balance here is though, I can see bumping this up to maybe 2 or 3 seconds. 

Does anyone have a strong opinion here?

Reply:
> resulting in about 5 or 6 attempts

If I understand correctly, this means around 5-6 seconds before a successful response (i.e. 5-6 tries, 1 per second)? I think it sounds reasonable, though I'm curious about whether there are multiple kinds of errors fall into the category. Also, curious about the generation step and / or cache miss. Is it a timeout issue? If so, can anything be done on the other end to keep it alive?

Reply:
> there are multiple kinds of errors fall into the category.

One error that could trigger these conditions would be if the CDN or service is malfunctioning. We saw that with one server while testing. They added some uptime logging to better catch that case. From the return result though it looked exactly the same as the "expected" failure case. 

>Is it a timeout issue? If so, can anything be done on the other end to keep it alive?

That's a good question, @enejb was investigating other improvements to this service still but I'm not sure which paths he's explored. A hope would be that if we discovered those paths after a release then we could turn it on and use this as a backup in case that didn't work well.

Reply:
> One error that could trigger ...

So maybe we should add a retry cap at say 10 tries which would be excessive for what I'm seeing in testing but wouldn't keep pinging for a URL that won't load. Since this is done in a collection view cell we would probably also need some parent data object to keep track of the retry attempts, that way the counter doesn't restart if you scroll away and come back. WDYT?

Reply:
That seems like a reasonable way to mitigate hammering the server, but I wonder if it would be worth the added complexity in practice. I think the way you have it makes sense, and 1 second seems reasonable, IMO.

One thing I noticed while testing is that it seems _all_ the images load at once, which was surprising. First, a single ghost image "pulsed", then afer a few seconds, several ghost images "pulsed" for several seconds (maybe close to 10 seconds total), and finally, all the images loaded nearly at once (excepting the ones off-screen, which loaded only once scrolled into view).

My hunch is that the API side is where to focus for improving the timing, and I think your throttle amount is a good start, and can be adjusted if needed as usage (in testing) ramps up.

Reply:
> I wonder if it would be worth the added complexity in practice.

I agree I don't think this would be worth the effort either. When we found the issue in testing our backend teams added monitoring on the server to catch it. So if that issue were to occur again we'd be able to address it there. 

> One thing I noticed while testing is that it seems all the images load at once, which was surprising.

What you're seeing here is actually the progression of the API calls. I can break down what's happening if you're interested:

>First, a single ghost image "pulsed"

This is the app trying to call and get the list of available designs. We use a single cell as a placeholder to help indicate "loading"

> several ghost images "pulsed" for several seconds (maybe close to 10 seconds total)

This is we know what designs are available but we don't have the thumbnails yet. For 10 seconds that would mean the thumbnail was still be generated on the server. Then that's when you see the images load.

If we see issues while testing we could look into limiting the number of active image downloads but that's another one that I don't think is worth the effort IMO.

Reply:
Right, that confirms what I thought about the sequence of api calls. :+1: I guess I expected that the thumbnails would each load at varying times, progressively replacing the ghost placeholders, and wondered whether it was a coincidence that they all loaded at once (i.e. I'd have thought that once the URLs were requested in parallel, the generation step would have had varying response times for the different designs). In any case, certainly not a blocker here :smile: